packages/nimbus/src/components/date-range-picker/date-range-picker.dev.mdx

```diff
@@ -0,0 +1,402 @@
+
+## Component Requirements
+
+### Date Value Types
+
+All date values **must** use `@internationalized/date` types:
+- `CalendarDate` for date-only scenarios
+- `CalendarDateTime` for dates with times (no timezone)
+- `ZonedDateTime` for timezone-aware dates
+
+### Accessibility
+
+For basic usage, the DateRangePicker handles all accessibility requirements internally-**no additional ARIA attributes are required**. However, in specific contexts you should add:
+
+- **Form labels**: Add a `<label>` element or `aria-label` when the picker's purpose isn't clear from surrounding content:
+  ```tsx
+  <label htmlFor="report-dates">Report date range</label>
+  <DateRangePicker id="report-dates" />
+
+  {/* Or using aria-label */}
+  <DateRangePicker aria-label="Select event date range" />
+  ```
+
+**Keyboard Navigation:**
+
+The component supports full keyboard interaction:
+- `Tab` / `Shift+Tab`: Navigate between date segments and buttons
+- `Arrow keys`: Increment/decrement date segments and navigate calendar
+- `Enter` / `Space`: Open calendar, select dates
+- `Escape`: Close calendar popover
+
+## Common Patterns
+
+### Filtering Data by Date Range
+
+A common use case is filtering lists or tables by date range:
+
+```jsx-live-dev
+const App = () => {
+  const [dateRange, setDateRange] = useState(null);
+
+  // In a real application, you'd filter data based on dateRange
+  const appliedFilter = dateRange
+    ? `Showing data from ${dateRange.start} to ${dateRange.end}`
+    : 'No date filter applied';
+
+  return (
+    <Stack direction="column" gap="400">
+      <DateRangePicker
+        value={dateRange}
+        onChange={setDateRange}
+      />
+      <Text fontSize="sm" color="neutral.11">
+        {appliedFilter}
+      </Text>
+    </Stack>
+  );
+}
+```
+
+### Event Scheduling with Times
+
+For scheduling events or appointments, combine date and time selection:
+
+```jsx-live-dev
+const App = () => {
+  const [eventRange, setEventRange] = useState(null);
+
+  return (
+    <Stack direction="column" gap="400">
+      <Text fontWeight="500">Event Duration</Text>
+      <DateRangePicker
+        granularity="minute"
+        value={eventRange}
+        onChange={setEventRange}
+        hourCycle={12}
+      />
+      {eventRange && (
+        <Text fontSize="sm" color="neutral.11">
+          Event scheduled from {eventRange.start.toString()} to {eventRange.end.toString()}
+        </Text>
+      )}
+    </Stack>
+  );
+}
+```
+
+## Testing Your Implementation
+
+### Basic Rendering Tests
+
+Verify the component renders with expected elements:
+
+```tsx
+import { render, screen } from '@testing-library/react';
+import { DateRangePicker } from '@commercetools/nimbus';
+
+describe('DateRangePicker', () => {
+  it('renders start and end date inputs', () => {
+    render(<DateRangePicker />);
+
+    // Each date input has 3 segments (month, day, year)
+    const dateInputs = screen.getAllByRole('spinbutton');
```

Comment:
Is it odd to check for the presence of the date segment? I had to open the documentation site to double check that this is a real thing (it is, of course).